File: doc-examples/example-java/src/test/java/example/BlobFunctionTest.java
package example;

import example.support.BlobEventListener;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class BlobFunctionTest {

    @Test
    void copiesTheBlobAndPublishesAnEvent() {
        try (BlobFunction function = new BlobFunction()) {
            String result = function.copy("blob content");

            assertEquals("blob content", result);
            BlobEventListener listener = function.getApplicationContext().getBean(BlobEventListener.class);
            assertEquals(List.of("blob content"), listener.getReceived());
        }
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/BlobFunctionTest.kt
package example

import example.support.BlobEventListener
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class BlobFunctionTest {

    @Test
    fun copiesTheBlobAndPublishesAnEvent() {
        BlobFunction().use { function ->
            val result = function.copy("blob content")

            assertEquals("blob content", result)
            val listener = function.applicationContext.getBean(BlobEventListener::class.java)
            assertEquals(listOf("blob content"), listener.received)
        }
    }
}
